js/search/search.js

```diff
@@ -28,11 +30,50 @@ class Search {
         this.messageData = [];
         this.userId = userId;
         this.isRealTimeIndexing = false;
-        this.decompress(key);
+        this.getUserConfig(key);
         this.collector = makeBoundTimedCollector(this.checkIsRealTimeIndexing.bind(this),
             searchConfig.REAL_TIME_INDEXING_TIME, this.realTimeIndexing.bind(this));
     }
 
+    /**
+     * Checks config version before decompression
+     * @param key
+     */
+    getUserConfig(key) {
+        Utils.getSearchUserConfig(this.userId)
+            .then((config) => {
+                if (config.indexVersion === searchConfig.INDEX_VERSION) {
+                    /**
+                     * decompress passing false as 2nd arg creates
+                     * a new index without decrypting the old
+                     * index
+                     */
+                    this.decompress(key, false);
+                } else {
```

Comment:
can you remove `else` and return in the `if` condition? Cleaner looking code :)

Reply:
Done 👍